app/master/build.py

```diff
@@ -61,7 +63,7 @@ def __init__(self, build_request):
         self._project_type = None
         self._build_completion_lock = Lock()  # protects against more than one thread detecting the build's finish
 
-        self._all_subjobs_by_id = {}
+        self._all_subjobs_by_id = OrderedDict()
```

Comment:
If this isn't an OrderedDict, then the subjobs aren't guaranteed to be stored in the order we want or expect. 

```bash
❯ python
Python 3.4.5 (default, Jun  8 2017, 14:05:03) 
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> d = {}
>>> d['a'] = 1
>>> d['b'] = 1
>>> d['c'] = 1
>>> d
{'c': 1, 'b': 1, 'a': 1}
```

This is a problem when we want to limit the number of subjobs we show. If you have `offset=0` and `limit=10`, it's fair to assume you want to see the first 10 subjobs. If the subjobs aren't ordered, then we'd end up returning _some_ 10 subjobs instead of the first 10.